### ISSUE TITLE
The intersection of a fillpolly and a yolov5 classy

### ISSUE THREAD

**Issue by ozicmoi** [ultralytics/yolov5#11084]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I am trying to change the color of the rectangle when it enters the fillpoly field of the class in yolov5 at the intersection of a fillpolly and a yolov5 classy. But I have question marks about how I should do this.

`import torch
from PIL import Image
import cv2
import numpy as np
from shapely.geometry import Polygon

model = torch.hub.load('C:/Users/ozic/Desktop/ozic/car_projects/object_detection/yolov5oz', 'custom',path='C:/Users/ozic/Desktop/ozic/car_projects/object_detection/yolov5s.pt', source='local') 
cap = cv2.VideoCapture("polygon.mp4")

while (cap.isOpened()):
    while True:
        ret, frame = cap.read()
        if not ret:
            break

        frame = cv2.resize(frame, (500, 300))
        overlay = frame.copy()

        results = model(frame)
       
        for _, det in enumerate(results.xyxy[0]):
            box = det.detach().cpu().numpy()[:5]
            box = [int(x) for x in box]
            print(box)
            x1, y1, x2, y2, name = box  
            cropped = frame[y1:y2, x1:x2].copy()
            color = (0, 255, 255)
            
            # Polygon corner points coordinates
            pts = np.array([[14, 70], [175, 190],[220, 180],[450,300],[500,300], [500, 260],[230,150],[210,150] ,[35, 70]],np.int32)
            pts = pts.reshape((-1, 1, 2))
            color = (0, 0, 255)
            poly=cv2.fillPoly(frame, [pts],color)
            alpha=0.3
            new=cv2.addWeighted(poly, alpha, overlay, 1 - alpha, 0)

            if name==0:

                a=cv2.rectangle(new, (x1, y1), (x2, y2), (0, 0, 0), 2)
                resultt = poly.intersection(a)

                print("sonucc",resultt)
            else:
                new=cv2.rectangle(new, (x1, y1), (x2, y2), (0, 0, 255), 2)

            cv2.imshow('frame', new)

        if cv2.waitKey(1) & 0xFF == ord('q'):
            break
    cap.release()

`

**Comment by glenn-jocher** [comment]
@ozicmoi it looks like you are trying to change the color of the rectangle when it enters the fillpoly field of the class in YOLOv5. Your code seems to be using OpenCV for image manipulation and YOLOv5 for object detection. To achieve your goal, you may want to compute the intersection of the filled polygon and the rectangle using OpenCV's functions, such as `cv2.bitwise_and` or `cv2.bitwise_or`, to change the color where they intersect.

You can refer to the documentation for more information on using OpenCV for geometrical operations. If you have further questions or need assistance with YOLOv5 integration, feel free to ask.
